packages/dht/src/connection/WebSocket/AutoCertifierClientFacade.ts

```diff
@@ -0,0 +1,77 @@
+import { AutoCertifierClient, SessionIdRequest, SessionIdResponse, CertifiedSubdomain, Certificate } from '@streamr/autocertifier-client'
+import { ListeningRpcCommunicator } from '../../exports'
+import { Logger, waitForEvent3 } from '@streamr/utils'
+
+const START_TIMEOUT = 60 * 1000
+
+const defaultAutoCertifierClientFactory = (
+    filePath: string,
+    autocertifierUrl: string,
+    autocertifierRpcCommunicator: ListeningRpcCommunicator,
+    wsServerPort: number
+) => new AutoCertifierClient(
+    filePath,
+    wsServerPort,
+    autocertifierUrl, (_, rpcMethodName, method) => {
+        autocertifierRpcCommunicator.registerRpcMethod(
+            SessionIdRequest,
+            SessionIdResponse,
+            rpcMethodName,
+            method
+        )                        
+    })
+
+export interface IAutoCertifierClient {
+    start(): Promise<void>
+    stop(): void
+    on(eventName: string, cb: (subdomain: CertifiedSubdomain) => void): void
+}
+
+interface AutoCertifierClientFacadeConfig {
+    autocertifierUrl: string
+    autocertifiedSubdomainFilePath: string
+    autocertifierRpcCommunicator: ListeningRpcCommunicator
+    wsServerPort: number
+    setHost: (host: string) => void
+    updateCertificate: (certificate: Certificate) => void
+    createClientFactory?: () => IAutoCertifierClient
+}
+
+const logger = new Logger(module)
+
+export class AutoCertifierClientFacade {
+
+    private autocertifierClient: IAutoCertifierClient
+    private readonly setHost: (host: string) => void
+    private readonly updateCertificate: (certificate: Certificate) => void
```

Comment:
Are these some kind of event listeners? Would it make sense to have just one event callback, e.g. `onSubdomainUpdated`?

Reply:
These are functions passed for the AutoCertifierClientFacade that are use to update the state of the WebSocketServer but they could be a single function